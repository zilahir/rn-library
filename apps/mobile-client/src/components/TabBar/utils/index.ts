export { default as icons } from "./icons";

export { onPress } from "./navigations";
