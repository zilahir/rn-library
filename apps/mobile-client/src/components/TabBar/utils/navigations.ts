import { BottomTabBarProps } from "@react-navigation/bottom-tabs";

interface INavigationProps {
  navigation: BottomTabBarProps["navigation"];
  targetKey: string;
}

/**
 *
 * @param root0
 * @param root0.navigation
 * @param root0.target
 * @param root0.isFocused
 * @param root0.targetKey
 * @param root0.targetName
 */
export function onPress({
  navigation,
  targetKey,
  isFocused,
  targetName,
}: INavigationProps & { isFocused: boolean; targetName: string }): void {
  const event = navigation.emit({
    type: "tabPress",
    target: targetKey,
    canPreventDefault: true,
  });

  if (!isFocused && !event.defaultPrevented) {
    navigation.navigate(targetName);
  }
}

/*
export function onLongPress({ navigation, targetKey }: INavigationProps): void {
  navigation.emit({
    type: "tabLongPress",
    target: targetKey,
  });
}
*/
