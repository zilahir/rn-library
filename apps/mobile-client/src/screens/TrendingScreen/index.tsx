import { ReactElement } from "react";
import { View, Text } from "react-native";

/**
 * @returns {ReactElement} - react component
 * @description - renders trending screen
 */
function TrendingScreen(): ReactElement {
  return (
    <View>
      <Text>TrendingScreen</Text>
    </View>
  );
}

export default TrendingScreen;
