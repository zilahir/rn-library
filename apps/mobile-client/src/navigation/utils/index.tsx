import { ComponentProps } from "react";
import { BottomTabParamList } from "@app/navigation/types";
import HomeScreen from "@app/screens/HomeScreen";
import ProfileScreen from "@app/screens/ProfileScreen";
import TrendingScreen from "@app/screens/TrendingScreen";
import LibraryScreen from "@app/screens/LibraryScreen";

type ScreenName = keyof BottomTabParamList;

interface BottomTabScreen {
  label?: ScreenName;
  name: ScreenName;
  tabComponent: ComponentProps<any>;
  hideLabel?: boolean;
  isFloating?: boolean;
}

interface IBottomTabScreens {
  screens: BottomTabScreen[];
  getBottomTabBarTabs: () => BottomTabScreen[];
  getNumberOfTabBarTabs: () => number;
}

export const bottomTabScreens: IBottomTabScreens = {
  screens: [
    {
      label: "Home",
      name: "Home",
      tabComponent: HomeScreen,
    },

    {
      label: "Profile",
      name: "Profile",
      tabComponent: ProfileScreen,
    },
    {
      label: "Trending",
      name: "Trending",
      tabComponent: TrendingScreen,
    },
    {
      label: "Library",
      name: "Library",
      tabComponent: LibraryScreen,
    },
  ],
  getBottomTabBarTabs: (): BottomTabScreen[] =>
    bottomTabScreens.screens.map((screen) => ({
      ...screen,
      hideLabel: !!screen.hideLabel,
      isFloating: !!screen.isFloating,
    })),

  getNumberOfTabBarTabs: (): number => bottomTabScreens.screens.length,
};
