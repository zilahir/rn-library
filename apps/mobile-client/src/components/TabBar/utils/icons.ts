import { Feather } from "@expo/vector-icons";
import { ComponentProps } from "react";
import { BottomTabParamList } from "@app/navigation/types";

const icons: Record<
  keyof BottomTabParamList,
  ComponentProps<typeof Feather>["name"]
> = {
  Home: "home",
  Profile: "user",
  Trending: "trending-up",
  Library: "book-open",
  Borrow: "book",
};

export default icons;
