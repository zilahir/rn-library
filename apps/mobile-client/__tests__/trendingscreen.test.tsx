import TrendingScreen from "@app/screens/TrendingScreen";

import { render } from "./utils/wrapper";

describe("TrendingScreen", () => {
  it("render correctly", () => {
    const { toJSON } = render(<TrendingScreen />);

    expect(toJSON()).toMatchSnapshot();
  });
});
