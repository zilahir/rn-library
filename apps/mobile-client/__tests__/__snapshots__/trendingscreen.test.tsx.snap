// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendingScreen render correctly 1`] = `
<View>
  <Text>
    TrendingScreen
  </Text>
</View>
`;
